pkg/cluster/cluster.go

```diff
@@ -148,6 +151,10 @@ func (c *Cluster) create() error {
 		return fmt.Errorf("failed to update cluster phase (%v): %v", spec.ClusterPhaseCreating, err)
 	}
 
+	if err := c.gc.CollectCluster(c.cluster.Name, c.cluster.UID); err != nil {
+		return err
```

Comment:
this will mark the cluster failed, right? so the error will be user facing. we probably should wrap around it to make it more readable.